playwright/src/test/java/com/microsoft/playwright/junit/TestTracesAndScreenshots.java

```diff
@@ -0,0 +1,201 @@
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.microsoft.playwright.TestOptionsFactories.getBrowserName;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+
+// Test both traces and screenshots in the same class as they both have the same default directory
+// and cleanup is easier
+@Tag("fixtureTracing") // These tests should be run in isolation as there are threading issues when run with others
+@FixtureTest
+public class TestTracesAndScreenshots {
+  // Select all the test classes that need to be executed as part of this test
+  private static final EngineTestKit.Builder engineTestKitBuilder = EngineTestKit.engine("junit-jupiter")
+    .selectors(selectClass(TestTraceWithCustomOutputPath.class), selectClass(TestTraceOn.class), selectClass(TestTraceOff.class),
+      selectClass(TestTraceRetainOnFailures.class), selectClass(TestScreenshotOn.class), selectClass(TestScreenshotOff.class),
+      selectClass(TestScreenshotOnlyOnFailures.class), selectClass(TestScreenshotWithCustomOutputPath.class));
+
+  @BeforeAll
+  static void beforeAll() {
+    cleanUpDirs();
+    // run tests
+    engineTestKitBuilder.execute();
+  }
+
+  @AfterAll
+  static void afterAll() {
+    cleanUpDirs();
+  }
+
+  @Test
+  void testTraceWithCustomOutputPath() {
+    Class<?> testClass = TestTraceWithCustomOutputPath.class;
+    List<Method> testMethods = getTestMethods(testClass);
+
+    for (Method test : testMethods) {
+      File traceFile = getTracePathForCustomLocation(getCustomOutputPath(), testClass, test).toFile();
+      assertTrue(traceFile.exists());
+      assertTrue(traceFile.length() > 0);
+    }
+  }
+
+  @Test
+  void testTraceOn() {
+    Class<?> testClass = TestTraceOn.class;
+    List<Method> testMethods = getTestMethods(testClass);
+
+    for (Method test : testMethods) {
+      File traceFile = getTracePathForDefaultLocation(testClass, test).toFile();
+      assertTrue(traceFile.exists());
+      assertTrue(traceFile.length() > 0);
+    }
+  }
+
+  @Test
+  void testTraceOff() {
+    Class<?> testClass = TestTraceOff.class;
+    List<Method> testMethods = getTestMethods(testClass);
+
+    for (Method test : testMethods) {
+      File traceFile = getTracePathForDefaultLocation(testClass, test).toFile();
+      assertFalse(traceFile.exists());
+    }
+  }
+
+  @Test
+  void testTraceRetainOnFailure() {
+    Class<?> testClass = TestTraceRetainOnFailures.class;
+    List<Method> testMethods = getTestMethods(testClass);
+
+    for (Method test : testMethods) {
+      File traceFile = getTracePathForDefaultLocation(testClass, test).toFile();
+      if (test.getName().contains("ShouldNot")) {
```

Comment:
I'd check entire method name "traceShouldNotExistWhenTestPasses"  for clarity.

Also here and in other tests, when iterating over several methods, include the name in the message for easier debugging: assertFalse(traceFile.exists(), test.getName());